src/components/CollaboratorModal/components/MainSubmodal.tsx

```diff
@@ -0,0 +1,251 @@
+  const { siteName } = useParams<{ siteName: string }>()
+  const [showAck, setShowAck] = useState(false)
+  const successToast = useSuccessToast()
+  const {
+    mutateAsync: addCollaborator,
+    error: addCollaboratorError,
+    isSuccess: addCollaboratorSuccess,
+    isError: isAddCollaboratorError,
+    isLoading: isAddCollaboratorLoading,
+    reset,
+  } = CollaboratorHooks.useAddCollaboratorHook(siteName)
+  const { data: role } = CollaboratorHooks.useGetCollaboratorRoleHook(siteName)
+
+  const errorMessage = extractErrorMessage(
+    addCollaboratorError?.response?.data.error
+  )
+
+  const collaboratorFormMethods = useForm({
+    mode: "onTouched",
+    defaultValues: {
+      newCollaboratorEmail: "",
+      isAcknowledged: false,
+    },
+  })
+
+  // Show acknowledgement modal if user does not have a trusted email
+  useEffect(() => {
+    // No error message implies no error
+    if (!addCollaboratorError?.response?.data?.error) return
+
+    const { message: errMessage } = addCollaboratorError?.response?.data?.error
+
+    if (errMessage === ACK_REQUIRED_ERROR_MESSAGE) {
+      setShowAck(true)
+    }
+  }, [addCollaboratorError])
+
+  useEffect(() => {
+    if (addCollaboratorSuccess) {
+      successToast({ description: "Collaborator added successfully" })
+      collaboratorFormMethods.reset()
+    }
+  }, [addCollaboratorSuccess, collaboratorFormMethods, successToast])
+
+  const isDisabled = role !== "ADMIN"
+
+  return (
+    <Modal
+      onCloseComplete={() => {
+        reset()
+        setShowAck(false)
+        props.onCloseComplete?.()
+      }}
+      {...props}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <FormProvider {...collaboratorFormMethods}>
+          <form
+            onSubmit={collaboratorFormMethods.handleSubmit(async (data) => {
+              await addCollaborator(data)
+              // This could have been reached after acknowledgement
+              // if the added person is not whitelisted.
+              setShowAck(false)
+            })}
+          >
+            <ModalHeader>
+              {showAck
+                ? "Acknowledge Terms of Use to continue"
+                : "Manage collaborators"}
+            </ModalHeader>
+            <ModalCloseButton />
+            <ModalBody mb="3.25rem">
+              {showAck ? (
+                <AcknowledgementSubmodalContent
+                  isLoading={isAddCollaboratorLoading}
+                />
+              ) : (
+                <FormControl
+                  isRequired
+                  isInvalid={isAddCollaboratorError}
+                  isDisabled={isDisabled}
+                >
+                  <FormLabel>
+                    Only admins can add or remove collaborators
+                  </FormLabel>
+                  <Input
+                    {...collaboratorFormMethods.register(
+                      "newCollaboratorEmail"
+                    )}
+                  />
```

Comment:
Needs a placeholder text: `name@email.com`.

Also, I think this input should be cleared onClose, so that when the user reopens the modal it starts from a fresh state.

Reply:
good point wrt placeholder; wrt clearing - i clear only if the submission is successful. my thinking was that if the user mistyped, they'd rather have the old value there and just edit a few chars rather than having to re-enter. wdyt about this?

Reply:
Yeah that makes sense too, let's go with the original behaviour.

Reply:
The placeholder text still needs to be added hmm

Reply:
the placeholder disappeared in the rebase lmao